salt/sdb/rest.py

```diff
@@ -98,12 +98,15 @@ def query(key, value=None, service=None, profile=None):  # pylint: disable=W0613
     '''
     Get a value from the REST interface
     '''
-    comps = key.split('?')
-    key = comps[0]
-    key_vars = {}
-    for pair in comps[1].split('&'):
-        pair_key, pair_val = pair.split('=')
-        key_vars[pair_key] = pair_val
+    try:
+        key, query_string = key.split('?')
+        key_vars = {}
+        if query_string:
+           for pair in query_string.split('&'):
+               pair_key, pair_val = pair.split('=')
+               key_vars[pair_key] = pair_val
+    except ValueError:
```

Comment:
do you need this? 

why not use urllib and get rid of the custom parsing?

Reply:
Not sure I've changed nothing here, it was before :)

Reply:
I asked about the try/except? That wasn't there before, was it? Why do you need them?

But as I suggested earlier, why not replace the entire functionality custom with few urllib calls?

Reply:
Today we raise an error when the given url have no query string

Reply:
So, what inside the try/except you added will raise the ValueError you are trying to handle?

Reply:
I have the same question at @rares-pop here. I'm failing to see how this `except` ever gets hit. This sort of looks like something that was written to catch what would have been an `IndexError` in the old code, perhaps?

Reply:
Yes it is only to fix when you give a SDB URL without a query string.
I'll try to migrate this module to parse the URL via urllib if I have some times at work this week.

Reply:
Aren't you protected from that by this: if query_string: on line 104?

Reply:
Today I've a stack trace so no :)

Reply:
I think there is a simpler way to handle this, rather than wrapping the whole thing in a try/except, which can be difficult to reason around when wrapping blocks of code like this. For example, the issue referenced above is looking for an `IndexError`, not a `KeyError`. I'd rather we be more precise about the try/except.

Instead, what about just checking for a `?` in the `key` first?